packages/cli/src/cmds/beacon/initBeaconState.ts

```diff
@@ -56,12 +57,15 @@ async function initAndVerifyWeakSubjectivityState(
     );
   }
 
-  // Instead of warning user of wss check failure, we throw because user explicity wants to use
-  // the checkpoint sync
-  ensureWithinWeakSubjectivityPeriod(config, anchorState, anchorCheckpoint);
+  // Throw error unless user explicitly asked not to, in testnets can happen that wss period is too small
+  // that even some epochs of non finalization can cause finalized checkpoint to be out of valid range
+  const wssCheck = wrapFnError(() => ensureWithinWeakSubjectivityPeriod(config, anchorState, anchorCheckpoint));
+  if (wssCheck.err !== null && opts.ignoreWeakSubjectivityCheck !== true) {
+    throw wssCheck.err;
+  }
```

Comment:
What about assigning `isWithinWeakSubjectivityPeriod` earlier? (not biggest fan of wrapping err (as it reads like go/rust but it makes it much more readable here compared to using try-catch)

```suggestion
  const wssCheck = wrapFnError(() => ensureWithinWeakSubjectivityPeriod(config, anchorState, anchorCheckpoint));
  const isWithinWeakSubjectivityPeriod = wssCheck.err === null;

  if (!isWithinWeakSubjectivityPeriod && !opts.ignoreWeakSubjectivityCheck) {
    throw wssCheck.err;
  }
```